stests/core/clx/utils.py

```diff
@@ -24,24 +28,60 @@ def get_client(src: typing.Union[Node, NodeIdentifier, NetworkIdentifier, Execut
         node = src
     elif isinstance(src, NodeIdentifier):
         node = cache.infra.get_node(src)
+    elif isinstance(src, Network):
+        node = cache.infra.get_node_by_network_id(src)
     elif isinstance(src, NetworkIdentifier):
         node = cache.infra.get_node_by_network_id(src)
     elif isinstance(src, ExecutionContext):
         node = cache.infra.get_node_by_run_context(src)
     else:
         raise ValueError("Cannot derive node from input source.")
+
     if not node:
         raise ValueError("Network nodeset is empty, therefore cannot dispatch a deploy.")
 
     logger.log(f"PYCLX :: connecting to node :: {node.network}:N-{str(node.index).zfill(4)} :: {node.host}:{node.port}")
 
     # TODO: get node id / client ssl cert.
-    return node, pyclx.CasperLabsClient(
+    return node, casperlabs_client.CasperLabsClient(
         host=node.host,
         port=node.port,
     )
 
 
+def get_client_contract_path(contract_type=ClientContractType) -> pathlib.Path:
+    """Returns a path to a client side contract.
+
+    :param contract_type: Type of contract to be deployed.
+
+    :returns: Path to a client side contract.
+    
+    """
+    return pathlib.Path(casperlabs_client.__file__).parent / contract_type.value
+
+
+def get_client_contract_hash(
+    client: casperlabs_client.CasperLabsClient,
+    account: Account,
+    bhash: str,
+    contract_name: str
+    ) -> str:
+    """Returns chain hash of a deployed client side conttract.
+
+    :param client: A python client instance.
+    :param account: Account to be associated with contract.
+    :param bhash: Hash of block which processed deploy.
+    :param contract_name: Name of smart contract (within wasm blob).
+
+    :returns: Deploy client side contract hash.
+    
+    """
+    q = client.queryState(bhash, account.public_key, "", keyType="address")
+    for nk in q.account.named_keys:
+        if nk.name == contract_name:
+            return nk.key.hash.hash.hex()
+
```

Comment:
Rather than returning `None` in case `contract_name` was not found in account's named_keys perhaps it would be better to throw an exception.

Reply:
Now raising a ValueError if not found.